### ISSUE TITLE
SemanticKITTI Mean IoU Discrepancy

### ISSUE THREAD

**Issue by munib94** [QingyongHu/RandLA-Net#53]
Hello,

I ran your code for the SemanticKITTI pre-processing, training and evaluation, but only achieved a mean IoU of ~34 during training. I followed the instructions, but do not know why the mean IoU is so low. Do you know why this is happening?

**System setup:**
OS: CentOS Linux 7
TensorFlow-gpu version: 1.11.0
Python version: 3.5
CUDA/cuDNN version: 9.0/7.4.1
GPU: Nvidia GPU GeForce GTX 1080

**Comment by QingyongHu** [comment]
Hi @munib94, thanks for your interest in our work!

It is weird. Have you change any parameters or do any modifications?

**Comment by munib94** [comment]
Hi,

No, I did not change any parameters. The only modification I made was to the `dataset_path` in `mainSemanticKITTI.py`. I re-cloned a new copy of the RandLA-Net repo and tried training again, but I am running into the same issue. Except this time, the mean IoU is 37.745 at epoch 20 and has not changed since the previous three epochs. I have shared my log training file from my first training attempt.

[log_train_SemanticKITTIval.txt](https://github.com/QingyongHu/RandLA-Net/files/4476899/log_train_SemanticKITTIval.txt)

**Comment by QingyongHu** [comment]
Hi, @munib94, can you please check if you have followed our instructions step by step, including data processing?

As you mentioned, the code should not be able to run if you use GTX1080 and didn't change any parameter, as it reaches the memory limitation of the GPU. The following is the log_file I have obtained today.
[log_train_SemanticKITTI_08.txt](https://github.com/QingyongHu/RandLA-Net/files/4487354/log_train_SemanticKITTI_08.txt)

**Comment by munib94** [comment]
Sorry, I meant to write that I used GTX1080 Ti with 11GB memory. I have both GTX1080 and GTX1080 Ti installed.

```
0a:00.0 VGA compatible controller: Matrox Electronics Systems Ltd. G200eR2 (prog-if 00 [VGA controller])
	Subsystem: Dell Device 04cf
	Flags: bus master, medium devsel, latency 64, IRQ 19, NUMA node 0
	Memory at d8000000 (32-bit, prefetchable) [size=16M]
	Memory at deffc000 (32-bit, non-prefetchable) [size=16K]
	Memory at de000000 (32-bit, non-prefetchable) [size=8M]
	Expansion ROM at <unassigned> [disabled]
	Capabilities: [dc] Power Management version 1
	Kernel driver in use: mgag200
	Kernel modules: mgag200

42:00.0 VGA compatible controller: NVIDIA Corporation GP104 [GeForce GTX 1080] (rev a1) (prog-if 00 [VGA controller])
	Subsystem: ASUSTeK Computer Inc. Device 859b
	Flags: bus master, fast devsel, latency 0, IRQ 76, NUMA node 1
	Memory at d4000000 (32-bit, non-prefetchable) [size=16M]
	Memory at 3bff0000000 (64-bit, prefetchable) [size=256M]
	Memory at 3bfee000000 (64-bit, prefetchable) [size=32M]
	I/O ports at dc80 [size=128]
	[virtual] Expansion ROM at d5000000 [disabled] [size=512K]
	Capabilities: [60] Power Management version 3
	Capabilities: [68] MSI: Enable+ Count=1/1 Maskable- 64bit+
	Capabilities: [78] Express Legacy Endpoint, MSI 00
	Capabilities: [100] Virtual Channel
	Capabilities: [250] Latency Tolerance Reporting
	Capabilities: [128] Power Budgeting <?>
	Capabilities: [420] Advanced Error Reporting
	Capabilities: [600] Vendor Specific Information: ID=0001 Rev=1 Len=024 <?>
	Capabilities: [900] #19
	Kernel driver in use: nvidia
	Kernel modules: nouveau, nvidia_drm, nvidia

43:00.0 VGA compatible controller: NVIDIA Corporation GP102 [GeForce GTX 1080 Ti] (rev a1) (prog-if 00 [VGA controller])
	Subsystem: eVga.com. Corp. Device 6593
	Flags: bus master, fast devsel, latency 0, IRQ 77, NUMA node 1
	Memory at d6000000 (32-bit, non-prefetchable) [size=16M]
	Memory at 3bfd0000000 (64-bit, prefetchable) [size=256M]
	Memory at 3bfce000000 (64-bit, prefetchable) [size=32M]
	I/O ports at cc80 [size=128]
	[virtual] Expansion ROM at d7000000 [disabled] [size=512K]
	Capabilities: [60] Power Management version 3
	Capabilities: [68] MSI: Enable+ Count=1/1 Maskable- 64bit+
	Capabilities: [78] Express Legacy Endpoint, MSI 00
	Capabilities: [100] Virtual Channel
	Capabilities: [250] Latency Tolerance Reporting
	Capabilities: [128] Power Budgeting <?>
	Capabilities: [420] Advanced Error Reporting
	Capabilities: [600] Vendor Specific Information: ID=0001 Rev=1 Len=024 <?>
	Capabilities: [900] #19
	Kernel driver in use: nvidia
	Kernel modules: nouveau, nvidia_drm, nvidia

```
I followed your instructions step by step, including data processing, and trained a second time on the GTX1080 Ti, but only reached mean IoU 38.743 by epoch 80.
[log_train_SemanticKITTI08.txt](https://github.com/QingyongHu/RandLA-Net/files/4487620/log_train_SemanticKITTI08.txt)

**Comment by QingyongHu** [comment]
Hi @munib94. 
I have no idea about this, I will try to check the code and give you feedback in the near future.

**Comment by hansen7** [comment]
I have facing the same issues on the mIoU on validation set during training:
<img width="1203" alt="image" src="https://user-images.githubusercontent.com/14752739/86839908-f15aa400-c099-11ea-804a-12a45905405e.png">

But I have find that the model that I trained from scratch, has the similar performance on the test dataset on the SemanticKITTI submission port
<img width="828" alt="image" src="https://user-images.githubusercontent.com/14752739/86839623-92952a80-c099-11ea-9115-7dd697b4cf52.png">

**Comment by munib94** [comment]
I just tested it on the test dataset and I did not achieve similar performance. The third result in the image below is the RandLA-Net network I trained from scratch.
![Screen Shot 2020-07-20 at 5 20 16 PM](https://user-images.githubusercontent.com/37734669/87987867-57d5be00-caad-11ea-81c3-a3f1e2612ec0.png)

**Comment by JerryIndus** [comment]
> Hi, @munib94, can you please check if you have followed our instructions step by step, including data processing?
> 
> As you mentioned, the code should not be able to run if you use GTX1080 and didn't change any parameter, as it reaches the memory limitation of the GPU. The following is the log_file I have obtained today.
> [log_train_SemanticKITTI_08.txt](https://github.com/QingyongHu/RandLA-Net/files/4487354/log_train_SemanticKITTI_08.txt)

Excuse me, have you already solved this problem, I am now facing the same issue... And I have no clue, looking forward to your reply.

**Comment by universitypengyuyan** [comment]
> 对不起，我的意思是写我使用GTX1080 Ti与11GB内存。我同时安装了 GTX1080 和 GTX1080 Ti。
> 
> ```
> 0a:00.0 VGA compatible controller: Matrox Electronics Systems Ltd. G200eR2 (prog-if 00 [VGA controller])
> 	Subsystem: Dell Device 04cf
> 	Flags: bus master, medium devsel, latency 64, IRQ 19, NUMA node 0
> 	Memory at d8000000 (32-bit, prefetchable) [size=16M]
> 	Memory at deffc000 (32-bit, non-prefetchable) [size=16K]
> 	Memory at de000000 (32-bit, non-prefetchable) [size=8M]
> 	Expansion ROM at <unassigned> [disabled]
> 	Capabilities: [dc] Power Management version 1
> 	Kernel driver in use: mgag200
> 	Kernel modules: mgag200
> 
> 42:00.0 VGA compatible controller: NVIDIA Corporation GP104 [GeForce GTX 1080] (rev a1) (prog-if 00 [VGA controller])
> 	Subsystem: ASUSTeK Computer Inc. Device 859b
> 	Flags: bus master, fast devsel, latency 0, IRQ 76, NUMA node 1
> 	Memory at d4000000 (32-bit, non-prefetchable) [size=16M]
> 	Memory at 3bff0000000 (64-bit, prefetchable) [size=256M]
> 	Memory at 3bfee000000 (64-bit, prefetchable) [size=32M]
> 	I/O ports at dc80 [size=128]
> 	[virtual] Expansion ROM at d5000000 [disabled] [size=512K]
> 	Capabilities: [60] Power Management version 3
> 	Capabilities: [68] MSI: Enable+ Count=1/1 Maskable- 64bit+
> 	Capabilities: [78] Express Legacy Endpoint, MSI 00
> 	Capabilities: [100] Virtual Channel
> 	Capabilities: [250] Latency Tolerance Reporting
> 	Capabilities: [128] Power Budgeting <?>
> 	Capabilities: [420] Advanced Error Reporting
> 	Capabilities: [600] Vendor Specific Information: ID=0001 Rev=1 Len=024 <?>
> 	Capabilities: [900] #19
> 	Kernel driver in use: nvidia
> 	Kernel modules: nouveau, nvidia_drm, nvidia
> 
> 43:00.0 VGA compatible controller: NVIDIA Corporation GP102 [GeForce GTX 1080 Ti] (rev a1) (prog-if 00 [VGA controller])
> 	Subsystem: eVga.com. Corp. Device 6593
> 	Flags: bus master, fast devsel, latency 0, IRQ 77, NUMA node 1
> 	Memory at d6000000 (32-bit, non-prefetchable) [size=16M]
> 	Memory at 3bfd0000000 (64-bit, prefetchable) [size=256M]
> 	Memory at 3bfce000000 (64-bit, prefetchable) [size=32M]
> 	I/O ports at cc80 [size=128]
> 	[virtual] Expansion ROM at d7000000 [disabled] [size=512K]
> 	Capabilities: [60] Power Management version 3
> 	Capabilities: [68] MSI: Enable+ Count=1/1 Maskable- 64bit+
> 	Capabilities: [78] Express Legacy Endpoint, MSI 00
> 	Capabilities: [100] Virtual Channel
> 	Capabilities: [250] Latency Tolerance Reporting
> 	Capabilities: [128] Power Budgeting <?>
> 	Capabilities: [420] Advanced Error Reporting
> 	Capabilities: [600] Vendor Specific Information: ID=0001 Rev=1 Len=024 <?>
> 	Capabilities: [900] #19
> 	Kernel driver in use: nvidia
> 	Kernel modules: nouveau, nvidia_drm, nvidia
> ```
> 
> 我按照您的指示一步一步地进行操作，包括数据处理，并在GTX1080 Ti上进行了第二次训练，但到epoch 80时才达到平均IoU 38.743。 [log_train_SemanticKITTI08.txt](https://github.com/QingyongHu/RandLA-Net/files/4487620/log_train_SemanticKITTI08.txt)

请问有解决吗？
